go-gin/main.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import "github.com/gin-gonic/gin"
+import "go-gin/prisma-client"
+
+func main() {
+
+	client := prisma.New(&prisma.PrismaOptions{
+		Debug:    true,
+		Endpoint: "http://localhost:4466/go-gin/dev",
+	})
+
+	r := gin.Default()
+
+	r.POST("/publish/:id", func(c *gin.Context) {
+		id := c.Param("id")
+		published := true
+		post, err := client.UpdatePost(&prisma.UpdatePostParams{
+			Where: &prisma.PostWhereUniqueInput{
+				ID: &id,
+			},
+			Data: &prisma.PostUpdateInput{
+				IsPublished: &published,
+			},
+		},
+		).Exec()
+
+		if err != nil {
+			panic(err)
+		}
+
+		c.JSON(200, gin.H{
+			"post": post,
+		})
+	})
+
+	r.POST("/delete/:id", func(c *gin.Context) {
+		id := c.Param("id")
+
+		post, err := client.DeletePost(&prisma.PostWhereUniqueInput{
+			ID: &id,
+		},
+		).Exec()
+
+		if err != nil {
+			panic(err)
+		}
+
+		c.JSON(200, gin.H{
+			"post": post,
+		})
+	})
+
+	r.POST("/draft", func(c *gin.Context) {
+		// TODO: How do we get request body?
```

Comment:
Is it [what](https://github.com/gin-gonic/gin/issues/1295) you are looking for ? 
```
buf := make([]byte, 1024)
num, _ := c.Request.Body.Read(buf)
reqBody := string(buf[0:num])
```

Reply:
yeah thanks!